lib/BC/PcodeCFG.cpp

```diff
@@ -0,0 +1,166 @@
+#include <remill/BC/PCodeCFG.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <sleigh/op.hh>
+#include <sleigh/opcodes.hh>
+#include <sleigh/pcoderaw.hh>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+#include "lib/Arch/Sleigh/ControlFlowStructuring.h"
+
+namespace remill {
+namespace sleigh {
+
+PcodeCFG
+PcodeCFGBuilder::CreateCFG(const std::vector<RemillPcodeOp> &linear_ops) {
+  return PcodeCFGBuilder(linear_ops).Build();
+}
+
+
+std::vector<size_t> PcodeCFGBuilder::GetBlockStarts() {
+  std::vector<size_t> res;
+  res.push_back(0);
+
+
+  for (size_t curr_index = 0; curr_index < this->linear_ops.size();
+       curr_index += 1) {
+    auto interproc = this->GetIntraProcTargets(curr_index);
+    res.insert(res.end(), interproc.begin(), interproc.end());
+  }
+  return res;
+}
+
+PcodeBlock PcodeCFGBuilder::BuildBlock(size_t start_ind, size_t next_start) {
+  std::vector<RemillPcodeOp> ops;
+
+  for (size_t i = start_ind; i < next_start; i++) {
+    ops.push_back(this->linear_ops[i]);
+  }
+
+  return PcodeBlock(start_ind, std::move(ops),
+                    this->GetBlockExitsForIndex(next_start - 1));
+}
+
+
+namespace {
+struct IntraProcTransferCollector {
+  std::vector<size_t> operator()(const InstrExit &ex) {
+    return std::vector<size_t>();
+  }
+
+  std::vector<size_t> operator()(const IntrainstructionIndex &ex) {
+    return std::vector<size_t>(1, ex.target_block_ind);
+  }
+
+
+  std::vector<size_t> operator()(const Exit &ex) {
+    return std::visit(*this, ex);
+  }
+
+  std::vector<size_t> operator()(const ConditionalExit &ex) {
+    std::vector<size_t> res = std::visit(*this, ex.true_branch);
+    std::vector<size_t> other = std::visit(*this, ex.false_branch);
+    res.insert(res.end(), other.begin(), other.end());
+    return res;
+  }
+};
+}  // namespace
+
+std::vector<size_t> PcodeCFGBuilder::GetIntraProcTargets(size_t index) {
+  std::variant<Exit, ConditionalExit> ex = this->GetBlockExitsForIndex(index);
```

Comment:
```suggestion
  auto ex = GetBlockExitsForIndex(index);
```

Reply:
[ed9069f](https://github.com/lifting-bits/remill/pull/645/commits/ed9069fac284272cf1778130ec1d4659dc80960c) and [39ffa7b](https://github.com/lifting-bits/remill/pull/645/commits/39ffa7b8f169bbc3b9b8d577cc8921855404249f)